onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -290,1831 +494,2266 @@ Performs element-wise binary {name} on 8 bit data types (with Numpy-style broadc
+        .Attrs(QAttentionAttributes)
+        .Inputs(QAttentionInputs)
+        .Outputs(QAttentionOutputs)
+        .TypeConstraints(QAttentionTypeConstraints)
+        .TypeAndShapeInferenceFunction(QAttentionTypeAndShapeInference),
+    ONNX_NAMESPACE::CxOpSchema("EmbedLayerNormalization", "", 0)
+        .SetDomain(kMSDomain)
+        .SinceVersion(1)
+        .SetSupportLevel(OpSchema::SupportType::EXPERIMENTAL)
+        .Attrs(EmbedLayerNormalizationAttributes)
+        .Inputs(EmbedLayerNormalizationInputs)
+        .Outputs(EmbedLayerNormalizationOutputs)
+        .TypeConstraints(EmbedLayerNormalizationTypeConstraints)
+        .TypeAndShapeInferenceFunction(EmbedLayerNormalizationTypeAndShapeInference),
+    ONNX_NAMESPACE::CxOpSchema("FastGelu", "", 0)
+        .SetDomain(kMSDomain)
+        .SinceVersion(1)
+        .SetSupportLevel(OpSchema::SupportType::EXPERIMENTAL)
+        .Inputs(FastGeluInputs)
+        .Outputs(FastGeluOutputs)
+        .TypeConstraints(FastGeluTypeConstraints)
+        .TypeAndShapeInferenceFunction(ONNX_NAMESPACE::propagateShapeAndTypeFromFirstInput),
+    ONNX_NAMESPACE::CxOpSchema("SkipLayerNormalization", "", 0)
+        .SetDomain(kMSDomain)
+        .SinceVersion(1)
+        .SetSupportLevel(OpSchema::SupportType::EXPERIMENTAL)
+        .Attrs(SkipLayerNormalizationAttributes)
+        .Inputs(SkipLayerNormalizationInputs)
+        .Outputs(SkipLayerNormalizationOutputs)
+        .TypeConstraints(SkipLayerNormalizationTypeConstraints)
+        .TypeAndShapeInferenceFunction(ONNX_NAMESPACE::propagateShapeAndTypeFromFirstInput),
+};
+
+void Register(const ONNX_NAMESPACE::CxOpSchema& cx_schema) {
+    ONNX_NAMESPACE::OpSchema schema(cx_schema.name_, "", 0);
+    schema
+        .SetDomain(cx_schema.domain_)
+        .SinceVersion(cx_schema.since_version_)
+        .SetSupportLevel(cx_schema.support_);
+    for (const auto& attr : cx_schema.attributes_) {
+        schema.Attr(attr.toAttribute());
+    }
+    for (size_t i = 0; i < cx_schema.inputs_.size(); ++i) {
+        const auto& input = cx_schema.inputs_[i];
+        schema.Input(i, input.name, "", input.type_str, input.param_option, input.is_homogeneous, input.min_arity);
+    }
+    for (size_t i = 0; i < cx_schema.outputs_.size(); ++i) {
+        const auto& output = cx_schema.outputs_[i];
+        schema.Output(i, output.name, "", output.type_str, output.param_option, output.is_homogeneous, output.min_arity);
+    }
+    for (const auto& constrain : cx_schema.type_constraints_) {
+        std::vector<std::string> allowed_types;
+        allowed_types.reserve(constrain.allowed_type_strs.size());
+        for (const char * type : constrain.allowed_type_strs) {
+            allowed_types.emplace_back(type);
+        }
+        schema.TypeConstraint(constrain.type_param_str, allowed_types, "");
+    }
+    schema.TypeAndShapeInferenceFunction(cx_schema.inference_function_);
+    ONNX_NAMESPACE::OpSchemaRegistry::OpSchemaRegisterOnce{schema};
 }
```

Comment:
This is the piece of code that does CxOpSchema -> OpSchema conversion for registration, and shouldn't be needed if onnx has native support for some constexpr OpSchema